src/core/types.ts

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export const isReactNode = (value: any): value is React.ReactNode => {
```

Comment:
Yup sounds good! I think what the function does is fairly straightforward given its name and signature (this is also a somewhat common pattern) but its expected usage isn't and I'll document this (I just like to be a little careful with comments because code can change quickly and comments can become outdated :) )